src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -839,19 +846,25 @@ private async Task DestroyActivation(ActivationData activationData, Cancellation
             }
             catch (Exception ex)
             {
-                this.logger.Warn(ErrorCode.Catalog_DeactivateActivation_Exception, $"Exception when trying to deactivation {activationData}", ex);
+                this.logger.LogWarning((int)ErrorCode.Catalog_DeactivateActivation_Exception, ex, "Exception when trying to deactivate {Activation}", activationData);
             }
             finally
             {
                 lock (activationData)
                 {
                     activationData.SetState(ActivationState.Invalid);
                 }
-                // Capture grainInstance since UnregisterMessageTarget will set it to null...
-                var grainInstance = activationData.GrainInstance;
-                UnregisterMessageTarget(activationData);
-                RerouteAllQueuedMessages(activationData, null, "Finished Destroy Activation");
-                await activationData.DisposeAsync();
+
+                try
+                {
+                    UnregisterMessageTarget(activationData);
+                    RerouteAllQueuedMessages(activationData, null, "Finished Destroy Activation");
+                    await activationData.DisposeAsync();
+                }
+                catch (Exception exception)
+                {
+                    this.logger.LogWarning(exception, "Exception during deactivation");
```

Comment:
I think we should try to have better warning message, like logging the `ActivationData` where we have it. I know that we have the exception here, but a less generic message could be also useful